automata/tm/tape.py

```diff
@@ -1,33 +1,35 @@
 #!/usr/bin/env python3
 """Classes and methods for working with Turing machine tapes."""
 
-import collections
+from dataclasses import dataclass
 from typing import Iterator, Sequence, Tuple
 
 from typing_extensions import Self
 
 from automata.tm.tm import TMDirectionT
 
 
-class TMTape(
-    collections.namedtuple("TMTape", ["tape", "blank_symbol", "current_position"])
-):
+@dataclass(frozen=True)
+class TMTape:
     """A Turing machine tape."""
 
+    __slots__ = ("tape", "blank_symbol", "current_position")
+
     tape: Tuple[str]
     blank_symbol: str
     current_position: int
 
-    def __new__(
-        cls, tape: Sequence[str], *, blank_symbol: str, current_position: int = 0
+    def __init__(
```

Comment:
Missing type annotations on some functions in this PR. Without this, mypy won't check the internals.

Reply:
@eliotwrobson Thanks for the catch. I pushed up a08100c42d32dfe9752270d49fdb9d2a8df76a46 where (I think) I got them all.

Reply:
Sg! Once we get everything in the library, it might be good to turn on the no untyped defs flag in mypy.

Reply:
@eliotwrobson I agree! But we'll first need to add types to tests.